test/lib/util/package_clause_generator_test.rb

```diff
@@ -113,102 +113,104 @@ def test_errata_ids
                                     :filter => @filter, :errata_id => "Foo2")
 
-      expected_errata = [{"id" => {"$in" => ["Foo1", "Foo2"]}}]
+      expected_errata = [erratum_arel[:errata_id].in(["Foo1", "Foo2"])]
       assert_errata_rules([foo_rule, goo_rule], expected_errata)
     end
 
     def test_errata_dates_default
-      from = Date.today.to_s
-      to = Date.today.to_s
+      from = Date.today
+      to = Date.today
+      types = ["bugfix", "enhancement", "security"]
 
       @filter = FactoryBot.create(:katello_content_view_erratum_filter, :content_view => @content_view)
       foo_rule = FactoryBot.create(:katello_content_view_erratum_filter_rule,
-                                    :filter => @filter, :start_date => from, :end_date => to,
-                                    :types => ["bugfix", "enhancement", "security"])
+                                    :filter => @filter, :start_date => from.to_s, :end_date => to.to_s,
+                                    :types => types)
+
+      types_query = erratum_arel[:errata_type].in(types)
+      date_query = erratum_arel[:updated].gteq(from).and(erratum_arel[:updated].lteq(to))
 
-      expected = [{"$and" => [{"updated" => {"$gte" => from.to_time.utc.as_json,
-                                             "$lte" => to.to_time.utc.as_json}},
-                              {"type" => { "$in" => ["bugfix", "enhancement", "security"]}}]}]
+      expected = [date_query.and(types_query)]
       assert_errata_rules([foo_rule], expected)
     end
 
     def test_errata_dates_issued
-      from = Date.today.to_s
-      to = Date.today.to_s
+      from = Date.today
+      to = Date.today
+      types = ["bugfix", "security"]
 
       @filter = FactoryBot.create(:katello_content_view_erratum_filter, :content_view => @content_view)
       foo_rule = FactoryBot.create(:katello_content_view_erratum_filter_rule,
                                     :date_type => ContentViewErratumFilterRule::ISSUED,
-                                    :filter => @filter, :start_date => from, :end_date => to,
-                                    :types => ["security", "bugfix"])
+                                    :filter => @filter, :start_date => from.to_s, :end_date => to.to_s,
+                                    :types => types)
 
-      expected = [{"$and" => [{"issued" => {"$gte" => from.to_time.utc.as_json,
-                                            "$lte" => to.to_time.utc.as_json}},
-                              {"type" => {"$in" => ["security", "bugfix"]}}]}]
+      types_query = erratum_arel[:errata_type].in(types)
+      date_query = erratum_arel[:issued].gteq(from).and(erratum_arel[:issued].lteq(to))
+
+      expected = [date_query.and(types_query)]
       assert_errata_rules([foo_rule], expected)
     end
 
     def test_errata_dates_updated
-      from = Date.today.to_s
-      to = Date.today.to_s
+      from = Date.today
```

Comment:
Like this stuff. Why keep writing it over and over instead of putting it in a private method?